knockknock/slack_sender.py

```diff
@@ -50,60 +55,174 @@ def wrapper_sender(*args, **kwargs):
                 master_process = True
 
             if master_process:
-                contents = ['Your training has started 🎬',
-                            'Machine name: %s' % host_name,
-                            'Main call: %s' % func_name,
-                            'Starting date: %s' % start_time.strftime(DATE_FORMAT)]
-                contents.append(' '.join(user_mentions))
-                dump['text'] = '\n'.join(contents)
-                dump['icon_emoji'] = ':clapper:'
+                notification = "Your training has started! 🎬"
+                if user_mentions:
+                    notification = _add_mentions(notification)
+
+                dump["blocks"] = _starting_message(
+                    func_name, host_name, notification, start_time
+                )
+                dump["text"] = notification
+                dump["icon_emoji"] = ":clapper:"
```

Comment:
I might be missing something on the block builder web demo: I couldn't display this emoji (error with this line it seems).

Reply:
Unfortunately I found some bad news about this on the [slack webhook docs](https://api.slack.com/messaging/webhooks)

> You cannot override the default channel (chosen by the user who installed your app), username, or icon when you're using Incoming Webhooks to post messages. Instead, these values will always inherit from the associated Slack app configuration.

I tried some workarounds for this, but it seems that updating the name/emoji are only possible using the new api, which would change the configuration of this sender.

As a result, I removed `dump` and the `channel` parameter from this file entirely

Reply:
I have difficulties understanding: where is your message sent then (on which channel)? is it encoded in the URL in an identifiable way?
As far as possible, I think it's nice to have an identifiable bot sending messages (instead of yourself to yourself), something that the current version is handling well.

Reply:
Is this (https://api.slack.com/bot-users#installing-bot) related?

Reply:
Since `knockknock` only uses the webhook sender for slack, the bot user is set when initialising the app, and the channel is selected specifically for each webhook (as below).

![Screenshot 2020-03-17 at 10 23 22 pm](https://user-images.githubusercontent.com/35203669/76904018-6c0b7380-689f-11ea-8453-ce5ede60b6a9.png)

Messages sent to this webhook are still posted _as the bot_, but only to one channel, and with the same username and profile image every time.  This functionality looks like it has been deprecated for webhooks, and is only possible with the new API.

I can update the slack sender to use the new API if you wish, but it will bring it further away from the refactoring you mentioned earlier and will add a dependency on the [python slack client](https://github.com/slackapi/python-slackclient)

Reply:
@VictorSanh How would you like to proceed?  I believe the choices are:

1. Forgo the profile image formatting
2. Update this sender with a bot user account, and use [`chat.postMessage`](https://api.slack.com/methods/chat.postMessage) to specify channel and profile image (and potentially more functionality).  This would also mean updating the documentation for the new slack token setup.